### PR TITLE
ENH, SIMD: add new universal intrinsics for floor/rint

### PR DESCRIPTION
**Pull request by howjmay**

### CODE REVIEW

File: numpy/core/tests/test_simd.py [hunk 1]
```diff
@@ -331,12 +331,13 @@ def test_square(self):
...
 
     @pytest.mark.parametrize("intrin, func", [("self.ceil", math.ceil),
-    ("self.trunc", math.trunc)])
+    ("self.trunc", math.trunc), ("self.floor", math.floor)])
     def test_rounding(self, intrin, func):
         """
         Test intrinsics:
             npyv_ceil_##SFX
             npyv_trunc_##SFX
+            npyv_floor##SFX
```
Comment (seiko2plus) [inline numpy/core/tests/test_simd.py:340]: we need a test case for `-0.0 == -0.0`, you have disabled the whole signed-zero tests within #20367

**Comment by charris** [comment]
close/reopen

**Comment by charris** [comment]
Is everyone happy with this? Tests are passing.

**Comment by mattip** [comment]
Like in the other SIMD PRs, since this is meant to improve performance it would be helpful to have a before/after benchmark run, and a description of any change in the size of_multiarray_umath.so.

**Comment by seiko2plus** [comment]
@mattip, this pr only implement universal intrinsics for those mentioned operations without using them within umath module. #20397 should put these new intrinsics into action, once this pr gets merged.

**Comment by mattip** [comment]
Thanks @HowJMay, @seiko2plus

**Pull request merged**